osbs/build/build_request.py

```diff
@@ -859,6 +859,13 @@ def render_pulp_pull(self):
         if not pulp_registry:
             logger.info("removing %s from request, requires pulp_registry", pulp_registry)
             self.dj.remove_plugin(phase, plugin)
+            return
+
+        # Verify that we have a secret
+        if self.spec.pulp_secret.value is None:
+            raise OsbsValidationException("Pulp registry specified with no secret")
```

Comment:
Secrets are not required for unsecured registries, which we still want to support. Maybe a simple warning should be sufficient?